omod/src/main/java/org/openmrs/module/stockmanagement/web/controller/StockManagementRestController.java

```diff
@@ -0,0 +1,117 @@
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+// import org.apache.http.HttpResponse;
+import org.openmrs.Order;
+import org.openmrs.Patient;
+import org.openmrs.PatientIdentifier;
+import org.openmrs.PatientIdentifierType;
+import org.openmrs.annotation.Authorized;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.PersonService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
+// import org.openmrs.ui.framework.SimpleObject;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.openmrs.module.stockmanagement.api.Privileges;
+import org.openmrs.module.stockmanagement.api.StockManagementService;
+import org.openmrs.module.stockmanagement.api.dto.RecordPrivilegeFilter;
+import org.openmrs.module.stockmanagement.api.dto.StockInventoryResult;
+import org.openmrs.module.stockmanagement.api.dto.StockItemInventory;
+import org.openmrs.module.stockmanagement.api.dto.StockItemInventorySearchFilter;
+import org.openmrs.module.stockmanagement.api.utils.SimpleObject;
+
+import org.springframework.web.bind.annotation.CrossOrigin;
+
+/**
+ * This class configured as controller using annotation
+ */
+@Controller
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/stockmanagement/metrics")
+@Authorized
+public class StockManagementRestController {
+	
+	/** Logger for this class and subclasses */
+	protected final Log log = LogFactory.getLog(getClass());
+	
+	/**
+     * end point for getting metrics for total out of stock items
+     * @param request
+     * @return
+     */
+    @CrossOrigin(origins = "*", methods = {RequestMethod.GET, RequestMethod.POST, RequestMethod.PUT, RequestMethod.DELETE, RequestMethod.OPTIONS})
```

Comment:
It seems weird to define a CORS policy here. Surely that's something that should be deferred to the application level.

Reply:
Also, if you _are_ defining a CORS policy here, it should only allow `GET`s and `HEAD`s, since that's the only operation actually supported.